javascript/react/gotchas/dangerously-set-innerhtml.md

```diff
@@ -32,28 +32,62 @@ aspects:
 
 React gives you the ability to insert *raw* HTML.
 
-Take into  consideration that this practice is definitely to be avoided because it makes your code vulnerable to *cross-site scripting* (XSS).
+Take into consideration that this practice is definitely to be avoided because it makes your code vulnerable to *cross-site scripting* (XSS[1]).
 
- React named the prop which is used to pass this data as `dangerouslySetInnerHTML`. A key is used to specify the text with `__html`:
-```
-dangerouslySetInnerHTML:
-{ __html: '<p>My HTML text </p>' }
+React named the prop which is used to pass this data as `dangerouslySetInnerHTML`. A key is used to specify the text with `__html`:
+
+```jsx
+dangerouslySetInnerHTML: {
+  __html: "<p>My HTML text </p>";
+}
 ```
-The recommended way of using `dangerouslySetInnerHTML` when needed is by creating an object containing only the key `__html` and your *sanitized* data as the value.
+
+The recommended way of using `dangerouslySetInnerHTML` when needed is by creating an object containing only the key `__html` and your *sanitized*[2] data as the value.
+
+```jsx
+function createMarkup() {
+  return { __html: "<p>My HTML text </p>" };
+}
 ```
-function createMarkup()
-{ return {__html: '<p>My HTML text </p>'};};
+
+The next step is to pass the `createMarkup()` function as a value for your `dangerouslySetInnerHTML` prop:
+
+```jsx
+<div
+  dangerouslySetInnerHTML={createMarkup()}
+/>
 ```
 
 This method will prevent incorrect rendering.
 
+Let's say that you mistakenly write the following code:
+
+```jsx
+<div
+  dangerouslySetInnerHTML={getUsername()}
+/>
+```
+
+Fortunately, this will not render because `getUsername()` would return a string, but `dangerouslySetInnerHTML` expects and object of the type `__html: ''`. By forcing you to write an object with a special `__html` property, it increases that chance that you will remember to *sanitize* the data. 
```

Comment:
```suggestion
Fortunately, this will not render because `getUsername()` would return a string, but `dangerouslySetInnerHTML` expects an object of the shape `{ __html: string }`. This design is on purpose. By forcing you to write an object with an unusually named `__html` property, React increases the chances that you will actually remember to *sanitize* the data.
```